### PR TITLE
feat(theme): adjust font-size in compact mode

### LINKED ISSUE #23015: Enhancement of compact theme
ref: https://www.yuque.com/docs/share/1f9849ba-f80a-4a9a-a960-ba48a6d4027d

Should update in `feature` branch at `4.2.0`.

- [ ] Font size
- [ ] Component padding
- [ ] Button
- [ ] Breadcrumb
- [ ] Dropdown
- [ ] Layout
- [ ] Pagination
- [ ] PageHeader
- [ ] Steps
- [ ] AutoComplete
- [ ] Checkbox
- [ ] Cascadar
- [ ] DatePicker
- [ ] Form
- [ ] InputNumber
- [ ] Input
- [ ] Mentions
- [ ] Rate
- [ ] Radio
- [ ] Switch
- [ ] Slider
- [ ] Select
- [ ] TreeSelect
- [ ] Transfer
- [ ] TimePicker
- [ ] Upload
- [ ] Avatar
- [ ] Badge
- [ ] Comment
- [ ] Collapse
- [ ] Card
- [ ] Calendar
- [ ] Descriptions
- [ ] Empty
- [ ] List
- [ ] Popover
- [ ] Statistic
- [ ] Tree
- [ ] Tooltip
- [ ] Timeline
- [ ] Tag
- [ ] Tabs
- [ ] Table
- [ ] Alert
- [ ] Drawer
- [ ] Modal
- [ ] Message
- [ ] Notification
- [ ] Progress
- [ ] Popconfirm
- [ ] Result
- [ ] Spin
- [ ] Skeleton
- [ ] Anchor

### PR DESCRIPTION
**Pull request by AshoneA**
Closes #23015

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

Closes #23015

### 💡 Background and solution

With compact theme mode, we adjust default font-size from 14px to 12px.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     adjust font-size in compact mode      |
| 🇨🇳 Chinese |       调整紧凑模式下默认的字体大小    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
feature branch

File: components/alert/style/index.less [hunk 1]
```diff
@@ -113,7 +113,7 @@
 
   &-with-description &-icon {
     position: absolute;
-    top: 16px;
+    top: @alert-icon-top;
```
Comment (afc163) [inline components/alert/style/index.less:116]: 这个值最好能根据标题大小和 alert padding top 算出来。

File: components/style/themes/default.less [hunk 2]
```diff
@@ -854,6 +855,7 @@
 @alert-with-description-padding: @padding-md - 1px 15px @padding-md - 1px 64px;
 @alert-icon-top: 8px + @font-size-base * @line-height-base / 2 - @font-size-base / 2;
 @alert-with-description-icon-size: 24px;
+@alert-with-description-icon-top: @padding-md;
```
Comment (afc163) [inline components/style/themes/default.less:858]: 这个 top 应该是可以根据 @alert-with-description-padding 算出来？

Comment (AshoneA) [reply components/style/themes/default.less]: 应该直接用垂直方向上的 `padding` 就好了。

**Review by zombieJ — APPROVED** [review]

**Comment by zpr1g** [comment]
从 v4.1.4 到 v4.2.4 ，紧凑模式下的字体和间距感觉有点太小了，应该怎么自定义配置回 4.1.4 的那种效果呢？

**Pull request merged**